### ISSUE TITLE
RFC: Changes to `rw upgrade` command

### ISSUE THREAD

**Issue by dac09** [redwoodjs/redwood#1961]
Suggestions:

- [ ] Change upgrade to be more like https://github.com/redwoodjs/redwood/commit/322b35e2266d6d5f0da9d25c492008137cd80090
- [ ] Move upgrade, into a separate package so we can run something like `npx redwood-upgrade`
- [ ] Proxy `yarn rw upgrade` to `npx redwood-upgrade`
- [ ] remove node_modules as part of upgrade
- [ ] Upgrade react version and resolutions change, if required

**Comment by Tobbe** [comment]
What's the benefit of doing npx vs rw?

**Comment by dac09** [comment]
> What's the benefit of doing npx vs rw?
Let's take a specific example
- You're on v0.26.2
- We change the upgrade script in v0.27.0, to perform additional tasks required for your app in v0.27.
- However, when you run yarn rw upgrade on your project, its still using the upgrade command from v0.26.2

tldr;
Catch 22: you need to upgrade, to get the latest upgrade script

**Comment by Tobbe** [comment]
Got it. But I don't like it.
I like having everything under rw. Makes it easier to discover and learn.

Can we have `rw upgrade` download the latest version of itself before running? Maybe even just have `rw upgrade` do something like `execa('npm redwood-upgrade')` could work?

**Comment by dac09** [comment]
Yep, it could! Adding to RFC

**Comment by peterp** [comment]
@Tobbe `npx` allows you to upgrade, install and run a command locally. It's like `yarn create` but for packages that don't start with create: `create-package-name`

I propose we keep it in our namespace, `npx @redwoodjs/upgrade`

**Comment by Tobbe** [comment]
@peterp Yes, I know what npx is, been using it for `npx create-react-app` for year :)  

I just didn't like having another command for our users to use/find/learn

Having it all under `yarn rw` lets users find it by just running `yarn rw --help` (or even just `yarn rw`). But as I suggested to Danny, we could simply proxy `yarn rw upgrade` to `npx @redwoodjs/upgrade` :)

**Comment by peterp** [comment]
Soz, I "mansplained!"

**Comment by dac09** [comment]
This has been partially implemented!
